lib/internal/Magento/Framework/CompiledInterception/Generator/CompiledInterceptor.php

```diff
@@ -0,0 +1,723 @@
+        $returnTypeValue = $returnType
+            ? ($returnType->allowsNull() ? '?' : '') . $returnType->getName()
+            : null;
+        if ($returnTypeValue === 'self') {
+            $returnTypeValue = $method->getDeclaringClass()->getName();
+        }
+        return [
+            'name' => ($method->returnsReference() ? '& ' : '') . $method->getName(),
+            'parameters' =>array_map([$this, '_getMethodParameterInfo'], $parameters),
+            'body' => implode("\n", $body),
+            'returnType' => $returnTypeValue,
+            'docblock' => ['shortDescription' => '{@inheritdoc}'],
+        ];
+    }
+
+    /**
+     * Get scope cases from config
+     *
+     * @param array $config
+     * @return array
+     */
+    private function getScopeCasesFromConfig($config)
+    {
+        $cases = [];
+        //group cases by config
+        foreach ($config as $scope => $conf) {
+            $caseStr = "\tcase '$scope':";
+            foreach ($cases as &$case) {
+                if ($case['conf'] == $conf) {
+                    $case['cases'][] = $caseStr;
+                    continue 2;
+                }
+            }
+            $cases[] = ['cases'=>[$caseStr], 'conf'=>$conf];
+        }
+        $cases[count($cases) - 1]['cases'] = ["\tdefault:"];
+        return $cases;
+    }
+
+    /**
+     * Generate array with plugin info
+     *
+     * @param CompiledPluginList $plugins
+     * @param string $code
+     * @param string $className
+     * @param array $allPlugins
+     * @param null|string $next
+     * @return mixed
+     */
+    private function getPluginInfo(CompiledPluginList $plugins, $code, $className, &$allPlugins, $next = null)
+    {
+        $className = $plugins->getPluginType($className, $code);
+        if (!isset($allPlugins[$code])) {
+            $allPlugins[$code] = [];
+        }
+        if (empty($allPlugins[$code][$className])) {
+            $suffix = count($allPlugins[$code]) ? count($allPlugins[$code]) + 1 : '';
+            $allPlugins[$code][$className] = [
+                'code' => $code,
+                'class' => $className,
+                'clean_name' => preg_replace("/[^A-Za-z0-9_]/", '_', $code . $suffix)
+            ];
+        }
+        $result = $allPlugins[$code][$className];
+        $result['next'] = $next;
+        return $result;
+    }
+
+    /**
+     * Get next set of plugins
+     *
+     * @param CompiledPluginList $plugins
+     * @param string $className
+     * @param string $method
+     * @param array $allPlugins
+     * @param string $next
+     * @return array
+     */
+    private function getPluginsChain(CompiledPluginList $plugins, $className, $method, &$allPlugins, $next = '__self')
+    {
+        $result = $plugins->getNext($className, $method, $next);
+        if (!empty($result[DefinitionInterface::LISTENER_BEFORE])) {
+            foreach ($result[DefinitionInterface::LISTENER_BEFORE] as $k => $code) {
```

Comment:
@joni-jones this method would have five arguments with some of them passed by reference. I think such an update would only decrease code readability.

Reply:
How it would have five arguments if in both `if` conditions it has four arguments and the difference only in `DefinitionInterface::LISTENER_BEFORE` and `DefinitionInterface::LISTENER_AFTER`.